openbas-framework/src/main/java/io/openbas/integrations/PayloadService.java

```diff
@@ -205,5 +209,19 @@ private <T extends Payload> void duplicateCommonProperties(
     duplicate.setTags(new HashSet<>(origin.getTags()));
     duplicate.setExternalId(null);
     duplicate.setCollector(null);
+    duplicate.setSource(Payload.PAYLOAD_SOURCE.MANUAL);
+    duplicate.setStatus(Payload.PAYLOAD_STATUS.UNVERIFIED);
+  }
+
+  public void deprecateNonProcessedPayloadsByCollector(
+      String collectorId, List<String> processedPayloadExternalIds) {
+    List<String> existingPayloadExternalIds =
+        payloadRepository.findAllExternalIdsByCollectorId(collectorId);
+    List<String> payloadExternalIdsToDeprecate = new ArrayList<>(existingPayloadExternalIds);
+    payloadExternalIdsToDeprecate.removeAll(processedPayloadExternalIds);
+    payloadRepository.setPayloadStatusByExternalIds(
+        String.valueOf(Payload.PAYLOAD_STATUS.DEPRECATED), payloadExternalIdsToDeprecate);
+    LOGGER.log(
+        Level.INFO, "Number of deprecated Payloads: " + payloadExternalIdsToDeprecate.size());
```

Comment:
I am not sure to understand this part.
You are removing the deprecated payload without duplicate it ?

Reply:
Your question is regarding all the lines or just line 220 and 221

```
// We start a list with only the existing Payloads ids
List<String> payloadExternalIdsToDeprecate = new ArrayList<>(existingPayloadExternalIds);
// We remove the ids from the Collector
// At the end we get only the diff, the external Payloads ids to set as deprecated
// [A, B, C ] -  [B, C] = [A]
payloadExternalIdsToDeprecate.removeAll(processedPayloadExternalIds);
```

I updated the code by introducing a method to indicate that we get the ids to deprecate and I replaced the diff obtention with streams
```
  public void deprecateNonProcessedPayloadsByCollector(
      String collectorId, List<String> processedPayloadExternalIds) {
    List<String> payloadExternalIds =
        payloadRepository.findAllExternalIdsByCollectorId(collectorId);
    List<String> payloadExternalIdsToDeprecate =
        getExternalIdsToDeprecate(payloadExternalIds, processedPayloadExternalIds);
    payloadRepository.setPayloadStatusByExternalIds(
        String.valueOf(Payload.PAYLOAD_STATUS.DEPRECATED), payloadExternalIdsToDeprecate);
    LOGGER.log(
        Level.INFO, "Number of deprecated Payloads: " + payloadExternalIdsToDeprecate.size());
  }

  private static List<String> getExternalIdsToDeprecate(
      List<String> payloadExternalIds, List<String> processedPayloadExternalIds) {
    return payloadExternalIds.stream()
        .filter(externalId -> !processedPayloadExternalIds.contains(externalId))
        .collect(Collectors.toList());
  }
```

Let me know if it is clearer @RomuDeuxfois

Reply:
It's clearer, thank you so much !